yt/fields/magnetic_field.py

```diff
@@ -62,7 +64,7 @@ def _magnetic_field_strength(field, data):
 
     def _magnetic_energy(field, data):
         B = data[ftype,"magnetic_field_strength"]
-        return 0.5*B*B/mag_factors[B.units.dimensions]
+        return 0.5*B*B/mag_factors(B.units.dimensions)
```

Comment:
I sometimes get confused by closures -- since `mag_factors` only has `ds` from the outer namespace, I think it might be safer to make it accept `ds` as an argument and put it at the outermost function level.